Source/SwiftLintFramework/Rules/RuleConfigurations/SortedImportsConfiguration.swift

```diff
@@ -0,0 +1,43 @@
+enum SortedImportsGroupingConfiguration: String {
+    /// Sorts import lines based on any import attributes (e.g. `@testable`, `@_exported`, etc.), followed by a case
+    /// insensitive comparison of the imported module name.
+    case attributes
+    /// Sorts import lines based on a case insensitive comparison of the imported module name.
+    case name
+
+    init(value: Any) throws {
```

Comment:
This initializer isn't needed. You may just call `SortedImportsGroupingConfiguration(rawValue:)`.

Reply:
I would still need to handle an unknown value for `grouping`, either ignoring it or throwing:

```
        if let rawGrouping = configuration["grouping"] as? String,
           let grouping = SortedImportsGroupingConfiguration(rawValue: rawGrouping) {
            self.grouping = grouping
        }
```

or

```
        if let rawGrouping = configuration["grouping"] as? String {
            if let grouping = SortedImportsGroupingConfiguration(rawValue: rawGrouping) {
                self.grouping = grouping
            } else {
                throw ConfigurationError.unknownConfiguration
            }
        }
```

Any preference?

Reply:
I prefer an error for an unknown value (option 2).